File: 219. Contains Duplicate II/main.py
class Solution:
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        d={}
        l=len(nums)
        for i in range(l):
            if nums[i] not in d.keys():
                d[nums[i]]=[i]
            else:
                d[nums[i]].append(i)
                
        for t in d.keys():
            if len(d[t])>1:
                for i in range(len(d[t])-1):
                    if d[t][i+1]-d[t][i]<=k:
                        return True
                
        return False

File: 219. Contains Duplicate II/main.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        map<int,int> m;
        int l=nums.size();
        for(int i=0;i<l;i++){
            if(m.count(nums[i])==0)
                m[nums[i]]=i;
            else{
                if(i-m[nums[i]]<=k)
                    return true;
                m[nums[i]]=i;
            }
                
        }
        return false;
    }
};
